pkg/mutation/mutators/assignmeta/assignmeta_mutator_test.go

```diff
@@ -0,0 +1,78 @@
+package assignmeta
+
+import (
+	"fmt"
+	"testing"
+
+	mutationsv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func newFoo(spec map[string]interface{}) *unstructured.Unstructured {
+	data := map[string]interface{}{
+		"apiVersion": "v1",
+		"kind":       "Foo",
+		"metadata": map[string]interface{}{
+			"name": "my-foo",
+		},
+	}
+	if spec != nil {
+		data["spec"] = spec
+	}
+	return &unstructured.Unstructured{Object: data}
+}
+
+func newAssignMetadataMutator(path string, value mutationsv1alpha1.AssignField) *Mutator {
+	m := &mutationsv1alpha1.AssignMetadata{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Foo",
+		},
+	}
+	m.Spec.Parameters.Assign = value
+	m.Spec.Location = path
+
+	m2, err := MutatorForAssignMetadata(m)
+	if err != nil {
+		panic(err)
+	}
+	return m2
+}
+
+func TestAssign(t *testing.T) {
+	tests := []struct {
+		name  string
+		obj   *unstructured.Unstructured
+		path  string
+		value mutationsv1alpha1.AssignField
+		fn    func(*unstructured.Unstructured) error
+	}{
+		{
+			name:  "metadata value",
+			path:  "metadata.labels.foo",
+			value: mutationsv1alpha1.AssignField{FromMetadata: &mutationsv1alpha1.FromMetadata{Field: mutationsv1alpha1.ObjName}},
+			obj:   newFoo(map[string]interface{}{}),
+			fn: func(u *unstructured.Unstructured) error {
```

Comment:
Having this splits the test logic, making reading the test confusing. My guess here is that for each test, we want to check that a specific field is set to a particular value. We could instead define `key` and `wantValue` for each test:
```
key: []string{"metadata", "labels", "foo"}
value: "myfoo"
```

Then, the check can be done in the test logic itself. For example:

```
t.Run(..., ...
  got, _, err := unstructured.GetNestedString(u.Object, tc.key)
  if err != nil { t.Fatal(err) }
  if got != tc.value { t.Fatal("got %v = %v, want %v", key, got, tc.value) }
})
```

This is much easier read and grok the logic of.